integration/test/Testlib/ModService.hs

```diff
@@ -52,17 +47,19 @@ import Testlib.HTTP
 import Testlib.JSON
 import Testlib.Printing
 import Testlib.ResourcePool
+import Testlib.Service
 import Testlib.Types
 import Text.RawString.QQ
 import Prelude
 
-withModifiedService ::
-  Service ->
-  -- | function that edits the config
-  (Value -> App Value) ->
-  (String -> App a) ->
-  App a
-withModifiedService srv modConfig = runCodensity $ withModifiedServices (Map.singleton srv modConfig)
+withModifiedBackend :: HasCallStack => ServiceOverrides -> (HasCallStack => String -> App a) -> App a
+withModifiedBackend overrides k =
+  startDynamicBackends
+    [overrides]
+    ( \domains -> do
+        [domain] <- pure domains
```

Comment:
```suggestion
    ( \[domain] -> do
```
Might have to add `-Wincomplete-uni-patterns`. If you don't like it, just use `head` instead?